core/src/main/java/org/elasticsearch/index/mapper/core/DateFieldMapper.java

```diff
@@ -126,6 +128,10 @@ public DateFieldMapper build(BuilderContext context) {
 
         protected void setupFieldType(BuilderContext context) {
             FormatDateTimeFormatter dateTimeFormatter = fieldType().dateTimeFormatter;
+            if (fieldType().dateTimeFormatter().equals(Defaults.DATE_TIME_FORMATTER)) {
```

Comment:
I moved this check inside of the version check, if the index has been created before version 2.0 - in that case, it cannot be the default of `strictOptionalDate` and thus we can safely assume to fall back. Does this make sense or did I still overlook something in that scenario?